input/full.go

```diff
@@ -116,5 +116,9 @@ func CollectFull(server state.Server, connection *sql.DB, globalCollectionOpts s
 	ts.CollectorConfig = getCollectorConfig(server.Config)
 	ts.CollectorPlatform = getCollectorPlatform(globalCollectionOpts, logger)
 
+	logSnapshotDisabled, logSnapshotDisabledReason := getCollectorLogSnapshotStatus(server.Config, globalCollectionOpts, ts.Settings)
+	ts.CollectorLogSnapshotDisabled = logSnapshotDisabled
+	ts.CollectorLogSnapshotDisabledReason = logSnapshotDisabledReason
```

Comment:
So this is not the right place to put this because from here we can only really report this, we can't actually control log processing. We need to be able to turn on/off the setup that happens [here](https://github.com/pganalyze/collector/blob/stop-log-processing-if-using-bad-log-settings/main.go#L211-L253) when this value changes, right?

Reply:
The code part you reference only runs once at start (or SIGHUP) time, and I'm thinking we might need something that can respond to changes in the config without requiring a collector reload.

My first idea was to handle it later on in the grant handling, which is where the server-side deactivation would take effect:

https://github.com/pganalyze/collector/blob/stop-log-processing-if-using-bad-log-settings/logs/stream/stream.go#L259

However, its probably better to handle this in a way that also avoids the overhead on the collector side (i.e. don't do any log parsing in such situations), which means we should track this in the persisted state (or similar) and then take action based on that in the log handling. The deactivation logic could go early in the `ProcessLogStream` and `DownloadLogsFromAllServers` functions.

Re: state - There is actually already the `PersistedLogState` struct, which would be a good fit for this I think. We just need to make sure to lock the `LogStateMutex` when we modify/access that struct. Note that we currently don't yet lock this mutex for stream-based logs (only for RDS log file downloads), so we'll need to add that logic to `ProcessLogStream`.

Reply:
>The code part you reference only runs once at start (or SIGHUP) time, and I'm thinking we might need something that can respond to changes in the config without requiring a collector reload.

Sure, I just meant that we'd need to re-run that subscription logic every time this changed--but I guess you're suggesting we keep the subscriptions active and just discard data, which should be reasonably cheap and sure is simpler. I sketched out doing this in the patch I just pushed. Still needs some cleanup and fixes (I think the way I ended up doing this the subscriptions are just naturally consumed as log collection is forced off, but for AWS I think we'd need to clear the marker--I'll look into that), and I broke actually reporting the state for now, but I think this approach makes sense?

Reply:
(I also refactored a couple of things to make it easier to follow; sorry if that confuses things)